### PR TITLE
Issue/1070 adding parallelproj tests

### PR DESCRIPTION
**Pull request by markus-jehl**
Added parallelproj tests to BlocksOnCylindrical tests, including a comparison between parallelproj and ray tracing on the projection of a single voxel.

### CODE REVIEW

File: src/recon_test/test_blocks_on_cylindrical_projectors.cxx [hunk 1]
```diff
@@ -53,6 +54,8 @@
 #include "stir/VoxelsOnCartesianGrid.h"
 #include "stir/recon_buildblock/ForwardProjectorByBin.h"
 #include "stir/recon_buildblock/ForwardProjectorByBinUsingProjMatrixByBin.h"
+#include "stir/recon_buildblock/Parallelproj_projector/ForwardProjectorByBinParallelproj.h"
+#include "stir/recon_buildblock/Parallelproj_projector/BackProjectorByBinParallelproj.h"
```
Comment (KrisThielemans) [inline src/recon_test/test_blocks_on_cylindrical_projectors.cxx:58]: will have to be guarded by #ifdef

File: src/recon_test/test_blocks_on_cylindrical_projectors.cxx [hunk 2]
```diff
@@ -38,6 +38,7 @@
 #include "stir/ProjDataInfoBlocksOnCylindricalNoArcCorr.h"
 #include "stir/ProjDataInfoGenericNoArcCorr.h"
 #include "stir/ProjDataInfoCylindricalNoArcCorr.h"
+#include "stir/ProjDataInfoCylindrical.h"
```
Comment (KrisThielemans) [inline src/recon_test/test_blocks_on_cylindrical_projectors.cxx:41]: implied by previous includes

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
there's more here than I hoped, so I haven't reviewed this properly.

**Comment by markus-jehl** [comment]
Daniel's tests I only modified in that I introduced the templates and reordered them to be consistently ordered, because I was getting a bit confused which one was which. So the only real changes are the templates and the new test.

**Review by KrisThielemans — COMMENTED** [review]
looks fine, and checks work!

I'm not sure though why you templated all functions in the projector type. As far as I can see, using a `shared_ptr<ForwardProjectorByBin>` would work as well. No templating required. Feel like changing that?

(In fact,there's no reason here to pass a `shared_ptr` and using `ForwardProjectorByBin&` would be slightly better class design here, but that kind of stuff doesn't matter in the test of course).

**Comment by markus-jehl** [comment]
True, didn't think of that! Let me change that quickly.

**Comment by markus-jehl** [comment]
Some images to illustrate results of projections (left: ray tracing; right: parallelproj)
image
image
image
voxel
image

**Comment by KrisThielemans** [comment]
Excellent. Thanks!

There's also `test_view_offset_root.cxx` but feel free to decline fixing that now. (It's more important for the TOF branch later, so we could do that then).

**Comment by markus-jehl** [comment]
That doesn't seem to use projectors, but work straight on `ProjMatrixByBinUsingRayTracing`. I wouldn't know how to test this with parallelproj without digging into the code for a bit (or you telling me ;-) ).

**Review by danieldeidda — APPROVED** [review]
looks good

**Pull request merged**